### PR TITLE
mgr/dashboard: Hide maintenance option on expand cluster

### PR DESCRIPTION
**Pull request by nizamial09**
Fixes: https://tracker.ceph.com/issues/56070
Signed-off-by: Nizamudeen A <nia@redhat.com>

Screenshot from 2022-06-16 00-20-54

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins retest this please

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.html [hunk 1]
```diff
@@ -80,7 +80,8 @@
           </div>
 
           <!-- Maintenance Mode -->
-          <div class="form-group row">
+          <div class="form-group row"
+               *ngIf="pageURL === 'hosts'">
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.html:84]: That's ok, but (personally) I think it's preferable that conditions express what's the underlying reason for doing something.

There are multiple ways:
* For example, in the cluster expansion workflow, this component will be child/grandchild of the cluster expansion component, so if you inject that component in the constructor you'll be able to check that condition which really express why this is hidden (`if parentIsClusterCreation then hide this`).
* Even better, you can extend the interface of the hosts page with a new argument: `hideMaintenance` or `deploymentStage`, etc...

IMHO all those checks (`if parentIsClusterCreation`, `if hideMaintenance`, `if deploymentStage`) better reflect the intention of this change.

**Review by epuertat — APPROVED** [review]

**Comment by pereman2** [comment]
jekins test dashboard cephadm

**Comment by pereman2** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.html (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.html]: done! can you guys please check it again @pereman2 @epuertat

**Review by pereman2 — APPROVED** [review]
works locally so lgtm

**Review by sunilangadi2 — APPROVED** [review]
changes LGTM.

**Comment by pereman2** [comment]
jenkins test api

**Comment by epuertat** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test api

**Pull request merged**